src/raftstore/coprocessor/split_observer.rs

```diff
@@ -26,35 +26,28 @@ pub struct SplitObserver;
 type Result<T> = StdResult<T, String>;
 
 impl SplitObserver {
-    fn on_split(&self, ctx: &mut ObserverContext, split: &mut SplitRequest) -> Result<()> {
-        if split.get_split_key().is_empty() {
-            return Err("split key is expected!".to_owned());
-        }
-
-        let mut key = match split.get_split_key().decode_bytes(false) {
-            Ok(x) => x,
-            Err(_) => return Ok(()),
-        };
-
-        // format of a key is TABLE_PREFIX + table_id + RECORD_PREFIX_SEP + handle + column_id
-        // + version or TABLE_PREFIX + table_id + INDEX_PREFIX_SEP + index_id + values + version
-        // or meta_key + version
-        // The length of TABLE_PREFIX + table_id is TABLE_PREFIX_KEY_LEN.
-        if key.starts_with(table::TABLE_PREFIX) && key.len() > table::TABLE_PREFIX_KEY_LEN
-            && key[table::TABLE_PREFIX_KEY_LEN..].starts_with(table::RECORD_PREFIX_SEP)
-        {
-            // row key, truncate to handle
-            key.truncate(table::PREFIX_LEN + table::ID_LEN);
-        }
-
-        let region_start_key = ctx.region().get_start_key();
-
-        let key = encode_bytes(&key);
-        if *key <= *region_start_key {
-            return Err("no need to split".to_owned());
+    fn on_split(&self, ctx: &mut ObserverContext, splits: &mut [SplitRequest]) -> Result<()> {
+        for split in splits {
+            if split.get_split_key().is_empty() {
+                return Err("split key is expected!".to_owned());
+            }
+
+            let mut key = match split.get_split_key().decode_bytes(false) {
+                Ok(x) => x,
+                Err(_) => continue,
```

Comment:
continue here means we will still use the key for the split?

Reply:
Yes.